### PR TITLE
Add API for main extension

### PR DESCRIPTION
**Pull request by deerob4**

### CODE REVIEW

**Comment by michaeljones** [comment]
It looks broadly ok to me. I'm not confident in reviewing it though.

I'm assuming some of the code that has been removed is stuff that was popping up or happening automatically that made sense for the stand alone extension but not when integrated? 

I'm not sure why the Api class has some methods that are just empty? We don't seem to be implementing an interface or anything so I'm not sure why they exist?

**Comment by andrerim** [comment]
> I'm not sure why the Api class has some methods that are just empty? We don't seem to be implementing an interface or anything so I'm not sure why they exist?

I'm not sure either, @deerob4 didn't say why. They are either not needed or this isn't actually finished, I will take a deeper look.

File: src/api.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import * as vscode from 'vscode';
```
Comment (trond-snekvik) [inline src/api.ts:1]: Needs license

File: src/api.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    buildConfig: vscode.Uri;
+}
+
+class Api {
+    public version = 1;
+
+    public activationCfg: {
+        kconfigRoot: vscode.Uri | undefined;
+        zephyrBoard: string | undefined;
+        west: string | undefined;
```
Comment (trond-snekvik) [inline src/api.ts:23]:
```suggestion
        west?: string;
```
same above

File: src/api.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    setZephyrBoard(board: string): void {
+        return zephyr.updateBoardFromName(board);
+    }
+
+    setKconfigRoot(appUri: vscode.Uri): void {
+        const root = kEnv.findRootFromApp(appUri);
+        kEnv.setConfig('root', root);
+    }
```
Comment (trond-snekvik) [inline src/api.ts:41]: The ZephyrBoard and KconfigRoot should be tied to the context, like in the devicetree api. Even though there's just support for one context at the moment, we'll need more in the future, each with their own environment

File: src/api.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+class Api {
+    public version = 1;
+
+    public activationCfg: {
+        kconfigRoot: vscode.Uri | undefined;
+        zephyrBoard: string | undefined;
+        west: string | undefined;
+    } = { kconfigRoot: undefined, zephyrBoard: undefined, west: undefined };
```
Comment (trond-snekvik) [inline src/api.ts:24]:
```suggestion
    } = { };
```

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Comment by trond-snekvik** [comment]
Ironically, I'm not registered as a member of this repo, so my review doesn't count :smile: Will have to wait for @wbober to come back, unless the rules for this repo doesn't mandate reviews

**Comment by wbober** [comment]
@trond-snekvik This should be fixed now.

**Comment by bencefr** [comment]
I don't seem to find use of the introduced context, it only seem to update global configuration. My assumption is that all references to configuration values must be changed to refer to values mandated by the context and only use the configuration as a global fallback, otherwise the context makes no sense. I'd like to hear more from @trond-snekvik .

**Review by bencefr — APPROVED** [review]

**Pull request merged**